### PR TITLE
Modernize build infra + updated nullability from SDK

### PR DESCRIPTION
**Pull request by ZacSweers**
- Gradle 8.0.2
- Update dependencies
- Version catalogs
- JVM target 8
- Kotlin 1.8.10
- Updated new nullability around AttributeSet in the SDK

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,55 +1,24 @@
...
-        includeGroupByRegex "org\\.jetbrains.*"
-        includeGroupByRegex "com\\.jakewharton.*"
-        includeGroupByRegex "com\\.squareup.*"
-        includeGroupByRegex "junit.*"
-        includeGroupByRegex "org\\.assertj.*"
-        includeGroupByRegex "org\\.mockito.*"
-      }
+
+    tasks.withType(JavaCompile) {
+      options.release = 8
```
Comment (chrisjenx) [inline build.gradle:21]: Do we want to target 11? As I thought that was the default of AGP 7.4?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+leakcanary = "2.10"
+mavenPublish = "0.25.1"
+
+[plugins]
+agp-application = { id = "com.android.application", version.ref = "agp" }
+agp-library = { id = "com.android.library", version.ref = "agp" }
+dokka = { id = "org.jetbrains.dokka", version.ref = "dokka" }
+kotlin-android = { id = "org.jetbrains.kotlin.android", version.ref = "kotlin" }
+kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin" }
+mavenPublish = { id = "com.vanniktech.maven.publish", version.ref = "mavenPublish" }
```
Comment (chrisjenx) [inline gradle/libs.versions.toml:15]: I don't think this is needed anymore, I just use the built in maven publish plugin now as AGP correctly supports it.

**Review by chrisjenx — APPROVED** [review]
Two notes about JDK and MavenPublish, looks good otherwise

File: gradle/libs.versions.toml (hunk 2 above)
Comment (ZacSweers) [reply gradle/libs.versions.toml:15]: This is still useful if you want all the bells and whistles (dokka, snapshots, sonatype support, etc)

File: build.gradle (hunk 1 above)
Comment (ZacSweers) [reply build.gradle:21]: 8 is still the default, 11 (and soon 17) are the requirement for the plugin, but not the compiled sources themselves

**Pull request merged**